impl/client/src/main/javascript/web/dojo/pentaho/common/Dialog.js

```diff
@@ -140,7 +149,8 @@ define([
         var btn = document.createElement("BUTTON");
         domClass.add(btn, "pentaho-button");
         btn.setAttribute( "id", "button"+j);
-        btn.innerHTML = DOMPurify.sanitize(this.buttons[j]);
+        // this is filled with the sanitized localized strings, however buttons can be override hence the double check
+        btn.innerHTML = this.getLocaleSanitized(this.buttons[j]);
```

Comment:
While `sanitize` is shorter/nicer, the name is too broad and could apply to HTML, JS String, CSS... And it also resolves a key to a localized string, which is also not obvious from the name. Being a method of dialog, you can argue it narrows down the possible meanings. You may be right.

I would prefer a name which leaves less room for guessing.

If nothing better comes up, the plain: `getSanitizedLocaleString(key)` or `getSafeLocaleString(key)` would do.

Reply:
Valid point and changes were made for `getSanitizedLocaleString(key)`.